lib/src/callable/built_in.dart

```diff
@@ -55,13 +55,36 @@ class BuiltInCallable implements Callable, AsyncBuiltInCallable {
   /// Returns the argument declaration and Dart callback for the given
   /// positional and named arguments.
   ///
-  /// Note that this doesn't guarantee that [positional] and [names] are valid
-  /// for the returned [ArgumentDeclaration].
+  /// If no exact match is found, finds the closest approximation. Note that this
+  /// doesn't guarantee that [positional] and [names] are valid for the returned
+  /// [ArgumentDeclaration].
   Tuple2<ArgumentDeclaration, _Callback> callbackFor(
-          int positional, Set<String> names) =>
-      _overloads.take(_overloads.length - 1).firstWhere(
-          (overload) => overload.item1.matches(positional, names),
-          orElse: () => _overloads.last);
+      int positional, Set<String> names) {
+    Tuple2<ArgumentDeclaration, _Callback> fuzzyMatch;
+    int minMismatchDistance;
+
+    for (var i = 0; i < _overloads.length; i++) {
+      var overload = _overloads[i];
+
+      // Ideally, find an exact match.
+      if (overload.item1.matches(positional, names)) return overload;
+
+      int mismatchDistance = overload.item1.arguments.length - positional;
```

Comment:
```suggestion
      var mismatchDistance = overload.item1.arguments.length - positional;
```

You don't need to type-annotate variables that can be inferred.

Reply:
Gotcha